language/e2e-testsuite/src/tests/verify_txn.rs

```diff
@@ -241,35 +243,33 @@ fn verify_simple_payment() {
         StatusCode::GAS_UNIT_PRICE_ABOVE_MAX_BOUND
     );
 
+    // Test for a max_gas_amount that is insufficient to pay the minimum fee.
+    // Find the minimum transaction gas units and subtract 1.
+    let mut gas_limit = gas_constants
+        .to_external_units(gas_constants.min_transaction_gas_units)
+        .get();
+    if gas_limit > 0 {
+        gas_limit -= 1;
+    }
+    // Calculate how many extra bytes of transaction arguments to add to ensure
+    // that the minimum transaction gas gets rounded up when scaling to the
+    // external gas units. (Ignore the size of the script itself for simplicity.)
+    let mut extra_txn_bytes = 0;
+    if gas_constants.gas_unit_scaling_factor > gas_constants.min_transaction_gas_units.get() {
+        extra_txn_bytes = gas_constants.large_transaction_cutoff.get()
+            + (gas_constants.gas_unit_scaling_factor / gas_constants.intrinsic_gas_per_byte.get());
+    }
```

Comment:
Nit (personal preference, so ignore if you prefer the current way): 
```suggestion
    let extra_txn_bytes = if gas_constants.gas_unit_scaling_factor > gas_constants.min_transaction_gas_units.get() {
        extra_txn_bytes = gas_constants.large_transaction_cutoff.get()
            + (gas_constants.gas_unit_scaling_factor / gas_constants.intrinsic_gas_per_byte.get());
    } else {
        0
    };
```

Reply:
I had considered that, but my preference is not at all strong. I don't mind changing it.